Dockerfile

```diff
@@ -27,28 +29,45 @@ RUN apt-get update && apt-get -y install \
         automake \
         libtool \
         libmosquitto-dev \
-        libwebsockets-dev \
         pkg-config \
+        git \
+        cmake \
         make \
-    && apt-get clean
+    && apt clean
 
-FROM build-env AS runner
+RUN mkdir -p /usr/local/src
+WORKDIR /usr/local/src/
+RUN git clone https://github.com/warmcat/libwebsockets.git libwebsockets
+WORKDIR /usr/local/src/libwebsockets
+RUN cmake -B build -S .
+RUN cd build && make && make install
+# install libs in /usr/local/lib ; configured in /etc/ld.so.conf.d/libc.conf
+# ENV LD_LIBRARY_PATH /usr/local/lib:${LD_LIBRARY_PATH}
+RUN ldconfig -v
 
 ENV MAKE_JOBS=8
-ENV OBUSPA_ARGS="-v4"
 
-# Copy in all of the code
-# Then compile, as root.
 COPY . /obuspa/
 RUN cd /obuspa/ && \
     autoreconf -fi && \
     ./configure && \
     make -j${MAKE_JOBS} && \
     make install
 
-# Then delete the code
-# that's no longer needed
-RUN rm -rf /obuspa
+FROM debian:stable AS build-release-stage
+RUN apt update && apt -y install \
+    libssl-dev \
+    libsqlite3-dev \
+    libcurl4-openssl-dev\
+    libmosquitto-dev
+
+WORKDIR /
+COPY --from=build-env /usr/local/lib/libwebsockets.* /usr/local/lib
```

Comment:
Is it really that safe to copy from something built on ubuntu:latest to debian:stable, and expect it to work ? It seems safer to have the same version of Linux for both. And I think that in general, people seem to prefer Ubuntu over debian, as more familiar.

Reply:
Hello Richard,

Thanks for your review comment. I used debian image for the final one with an assumption that its size would be smaller than ubuntu. However I just checked in docker hub that its reverse. Now coming to ubuntu vs. debian compatibility, my understanding is that ubuntu is debian based hence it should work and at my setup it did work indeed. I agree to be on the safer side it would be better to follow the same. Will modify and resubmit.

Regards